pkg/scheduler/internal/cache/cache.go

```diff
@@ -665,3 +665,16 @@ func (cache *schedulerCache) expirePod(key string, ps *podState) error {
 func (cache *schedulerCache) NodeTree() *NodeTree {
 	return cache.nodeTree
 }
+
+// GetNodeInfo returns cached data for the node 'id'.
+func (cache *schedulerCache) GetNodeInfo(nodeID string) (*v1.Node, error) {
+	cache.mu.Lock()
```

Comment:
I think you just need a `RLock` here

Reply:
Also, do we have any performance hit with this?

Reply:
Comment has been addressed.

Did I use the wrong approach to run the scheduler perf tests? I followed the instructions in https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/test/integration/scheduler_perf/README.md, when I ran it in the master and this branch, and it gives warnings like this, but it seems like that there is a performance improvement.

```go
// master branch
$  ./test-performance.sh
...
Warning: pod scheduling throughput for 3k pods was slow for an interval... Saw an interval with very low (51) scheduling rate!PASS
ok  	k8s.io/kubernetes/test/integration/scheduler_perf	39.598s

// current branch
$  ./test-performance.sh
...
Warning: pod scheduling throughput for 3k pods was slow for an interval... Saw an interval with very low (79) scheduling rate!PASS
ok  	k8s.io/kubernetes/test/integration/scheduler_perf	37.803s
```

Reply:
FYI: the guide is at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-scheduling/scheduler_benchmarking.md